homeassistant/components/elmo_alarm/__init__.py

```diff
@@ -0,0 +1,265 @@
+"""Support for e-connect Elmo alarm system."""
+from collections import namedtuple
+from datetime import timedelta
+import logging
+
+from elmo.api.client import ElmoClient
+from elmo.api.exceptions import PermissionDenied
+from urllib3.exceptions import HTTPError
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_call_later
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "elmo_alarm"
+
+CONF_VENDOR = "vendor"
+
+CONF_STATES = "states"
+CONF_ZONES = "zones"
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+DEFAULT_HOST = "https://connect.elmospa.com"
+
+SIGNAL_ZONE_CHANGED = f"{DOMAIN}.zone_changed"
+SIGNAL_INPUT_CHANGED = f"{DOMAIN}.input_changed"
+SIGNAL_ARMING_STATE_CHANGED = f"{DOMAIN}.arming_state_changed"
+
+ZONE_ARMED = 0
+ZONE_DISARMED = 1
+
+INPUT_ALERT = 1
+INPUT_WAIT = 0
+
+ZoneData = namedtuple("ZoneData", ["zone_id", "zone_name", "state"])
+InputData = namedtuple("InputData", ["input_id", "input_name", "state"])
+
+
+STATE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): vol.In(
+            [
+                STATE_ALARM_ARMED_HOME,
+                STATE_ALARM_ARMED_AWAY,
+                STATE_ALARM_ARMED_NIGHT,
+                STATE_ALARM_ARMED_CUSTOM_BYPASS,
+            ]
+        ),
+        vol.Required(CONF_ZONES): vol.All(cv.ensure_list),
+    }
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST, default=DEFAULT_HOST): cv.string,
```

Comment:
Use `cv.url` to make sure that it is a valid url:
```suggestion
                vol.Required(CONF_HOST, default=DEFAULT_HOST): cv.url,
```

Reply:
Done.